src/lib.rs

```diff
@@ -96,64 +96,62 @@ fn adjust_heap<T, F>(v: &mut [T], hole_index: usize, len: usize, is_less: &mut F
 where
     F: FnMut(&T, &T) -> bool,
 {
+    assert!(len <= v.len());
```

Comment:
maybe `assume!` is better

Reply:
Here are benchmark results with asserts replaced by https://doc.rust-lang.org/std/intrinsics/fn.assume.html on nightly, with the current patch as the baseline. Tbh, I don't understand how the results can be so mixed:

```
partial sort 10000 limit 20                                                                                                                                                                                                                      
                        time:   [4.7747 µs 4.7855 µs 4.7967 µs]                                                                                                                                                                                  
                        change: [-37.758% -37.562% -37.343%] (p = 0.00 < 0.05)                                                                                                                                                                   
                        Performance has improved.                                                                       
Found 5 outliers among 100 measurements (5.00%)                                                                         
  1 (1.00%) low mild                                  
  3 (3.00%) high mild                            
  1 (1.00%) high severe
                                                            
partial sort 10000 limit 200                                                                                            
                        time:   [17.611 µs 17.632 µs 17.658 µs]                                                         
                        change: [-6.8429% -6.6376% -6.4391%] (p = 0.00 < 0.05)                                          
                        Performance has improved.                                                                       
Found 7 outliers among 100 measurements (7.00%)                                                                         
  4 (4.00%) high mild                                                                                                   
  3 (3.00%) high severe                                   
                                                            
partial sort 10000 limit 2000                                                                                           
                        time:   [149.65 µs 149.82 µs 149.99 µs]                                                         
                        change: [+12.786% +13.039% +13.289%] (p = 0.00 < 0.05)                                          
                        Performance has regressed.                                                                      
Found 12 outliers among 100 measurements (12.00%)                                                                       
  4 (4.00%) low severe                                
  2 (2.00%) low mild                             
  6 (6.00%) high mild
                                                            
partial sort 10000 limit 10000                                                                                          
                        time:   [266.96 µs 267.74 µs 268.51 µs]                                                         
                        change: [+0.4154% +0.8963% +1.3612%] (p = 0.00 < 0.05)                                                                                                                                                                   
                        Change within noise threshold.
```

Reply:
https://godbolt.org/z/zh1897o46

assert! works. I don't know if `assume` work or not, goldbot did not support it.